toolchain/sem_ir/stringify_type.cpp

```diff
@@ -131,17 +131,11 @@ auto StringifyTypeExpr(const SemIR::File& outer_sem_ir, InstId outer_inst_id)
         break;
       }
       case CARBON_KIND(AssociatedEntityType inst): {
-        if (step.index == 0) {
-          out << "<associated ";
-          steps.push_back(step.Next());
-          push_inst_id(sem_ir.types().GetInstId(inst.entity_type_id));
-        } else if (step.index == 1) {
-          out << " in ";
-          steps.push_back(step.Next());
-          push_inst_id(sem_ir.types().GetInstId(inst.interface_type_id));
-        } else {
-          out << ">";
-        }
+        out << "<associated ";
+        push_string(">");
```

Comment:
I was able to infer from reading that these push statements are executed in reverse order (FILO), though I was confused for a minute :) Do you think we could add a comment on push_string and push_inst_id saying so to help future readers?

Reply:
Yeah, something in the name calling out would be nice as an alternative

Reply:
I talked about this with @zygoloid , and he suggested we could make `push_string`, `push_inst_id`, and `push_specific_id` into methods on a "work stack" object, which would hopefully address this concern. I would be happy to do that in a follow-up PR, because that work is largely independent of this PR.

Reply:
In the meantime, I've added some comments to the `push_`* functions about the order reversing.